metabase/metabase
# Include subpath to the URL during downloading question results

Closes https://github.com/metabase/metabase/issues/33244

### Description

Subpath (deploying the app at some route like https://website.com/metabase) is not a standard way of handing URL requests, so urls starting with `/` e.g. `/api/card/3` will be calculated by browser starting from the origin or `https://website.com`, which is not correct as metabase handles BE requests using URLs including subpath (`/metabase/api/card/3`).

To bypass this browser behaivour we identify the deployment location (subpath or not) and manually build links to the downloading resources, this PR covers question results

### How to verify

1. run your local instance (frontend and backend separately), go to the question and download results (csv, json, xlsx) - make sure they show correct result
2. go to admin settings -> change metabase URL from `localhost:3000` to `localhost:9090/metabase/`
3. stop your frontend server, keep backend server running
4. build js files with `yarn build`, open `localhost:3000` and make sure js files are loaded
5. clone https://github.com/iethree/metabase-subpath-proxy: e.g. `cd .. && git clone git@github.com:iethree/metabase-subpath-proxy.git`
6. go to the  e.g. `cd ./metabase-subpath-proxy` and run `docker-compose up` (or `docker compose up`)
7. now open `localhost:9090/metabase/` - notice `/` at the end of the line, it's required
8. go to the question and try to download results, make sure they show correct results

### Demo

https://www.loom.com/share/5e3a325deb2a4d78be81b39d44b18bf9

### Checklist
